tests/planner/test_plan.py

```diff
@@ -1,19 +1,71 @@
+from cgitb import enable
+from platform import node
 import networkx as nx
 import pytest
-from tests.planner.data import GRAPHS
+from tests.planner.data import GRAPHS, GRAPH_1, GRAPH_2
+from tests.planner.utils import to_graph, to_plan
 
 from fal.planner.plan import plan_graph
+from fal.cli.selectors import ExecutionPlan
 
 
 @pytest.mark.parametrize("graph_info", GRAPHS)
 def test_planner_sub_graphs(graph_info):
     graph = graph_info["graph"]
-    new_graph = plan_graph(graph)
+    new_graph = plan_graph(graph, to_plan(graph))
 
     subgraphs = [
         maybe_subgraph
         for maybe_subgraph in new_graph.nodes
         if isinstance(maybe_subgraph, nx.DiGraph)
     ]
+    assert {frozenset(subgraph.nodes) for subgraph in subgraphs} == graph_info[
+        "subgraphs"
+    ]
+
+
+@pytest.mark.parametrize("graph_info", GRAPHS)
+def test_planner_disabled_chunking(graph_info):
+    graph = graph_info["graph"]
+    new_graph = plan_graph(graph, to_plan(graph), enable_chunking=False)
+
+    subgraphs = [
+        maybe_subgraph
+        for maybe_subgraph in new_graph.nodes
+        if isinstance(maybe_subgraph, nx.DiGraph)
+    ]
+    assert len(subgraphs) == 0
+
+
+@pytest.mark.parametrize(
+    "selected_nodes", [[], ["A", "F"], ["A", "B", "C", "D", "E", "F"]]
+)
+def test_planner_execution_plan(selected_nodes):
+    graph = to_graph(GRAPH_1)
+    new_graph = plan_graph(graph, ExecutionPlan(selected_nodes, "<test>"))
+    assert list(new_graph.nodes) == selected_nodes
+
+
+@pytest.mark.parametrize(
+    "excluded_nodes, expected_subgraphs",
+    [
+        ([], {frozenset(("B1", "B2")), frozenset(("E", "F", "G"))}),
```

Comment:
this means: we exclude `nothing` and we only get `[B1, B2], [E, F, G]`? Maybe I am lost on how `pytest.mark.parametrize` works.

Reply:
But does it make sense to exclude nothing and get less than al nodes?

Reply:
These are not all the nodes, but rather found subgraphs (chunks).

Reply:
I am not understanding this, sorry if it is obvious. You are excluding nothing, but you expect the subgraphs (chunks) to **not** contain all of the nodes? I would have expected something like
```suggestion
        ([], {frozenset(("A", "C")), frozenset(("B", "B1", "B2")), frozenset(("SB1")), frozenset(("E", "F", "G"))}),
```

and many more. I mean, you have an assert `expected_subgraphs == ...` which means you are not expecting a subset of the subgraphs to match, you expect ALL of them.

Maybe the ExecutionPlan is built somehow I am not getting?

Reply:
> You are excluding nothing, but you expect the subgraphs (chunks) to not contain all of the nodes? 

Ah, I understand you. The thing is, `subgraphs` only contain 'chunks' that includes more than 2 nodes. 
https://github.com/fal-ai/fal/blob/09548da54fa6221ddda40126544656f110b91c8c/src/fal/planner/plan.py#L20